src/router/index.ts

```diff
@@ -7,21 +7,49 @@ const router: Router = createRouter({
   routes,
 });
 
-router.beforeEach(async (to: RouteLocationNormalized /*, from */) => {
-  const auth: AuthStore = useAuthStore();
+router.beforeEach(async (to: RouteLocationNormalized, _from: RouteLocationNormalized) => {
+  const authStore: AuthStore = useAuthStore();
 
-  await auth.initializeAuthStatus();
+  await authStore.initializeAuthStatus();
 
   // Redirect authenticated users trying to access the login page to the start page
-  if (to.path === '/' && auth.isAuthed) {
+  if (to.path === '/' && authStore.isAuthed) {
     return { path: '/start' };
   }
 
-  if (to.meta['requiresAuth'] && !auth.isAuthed) {
+  if (to.meta['requiresAuth'] && !authStore.isAuthed) {
     window.location.href = `/api/auth/login?redirectUrl=${to.fullPath}`;
     return false;
   }
 
+  if (to.meta['requiresPermission']) {
+    /* check if the user has person management permissions */
+    switch (to.meta['requiresPermission']) {
+      case 'personenverwaltung':
+        if (authStore.hasPersonenverwaltungPermission) {
+          return true;
+        }
+        return false;
+      case 'rollenverwaltung':
+        if (authStore.hasRollenverwaltungPermission) {
+          return true;
+        }
+        return false;
+      case 'schulverwaltung':
+        if (authStore.hasSchulverwaltungPermission) {
+          return true;
+        }
+        return false;
+      case 'klassenverwaltung':
+        if (authStore.hasKlassenverwaltungPermission) {
+          return true;
+        }
+        return false;
+      default:
+        return false;
+    }
+  }
```

Comment:
Die Reihenfolge sieht anders aus als was wir in der GUI haben. Wollen wir das vielleicht vereinheitlichen? 

Personenverwaltung => Klassen => Rollen => Schulen.

Wenn der User nur den Permission "rollenverwaltung" und "klassenverwaltung" hat dann landet er in die Rollenergebnisliste aber laut GUI ist das ein Schritt nach unten unter Klassenverwaltung. 

Wie siehst du das?

Reply:
Sehe ich genauso. Setze ich heute noch um